_posts/2025-03-31-zscore-hybrid-search.md

```diff
@@ -0,0 +1,170 @@
+---
+layout: post
+title:  Z Score Normalization Technique for Hybrid Search
+authors:
+  - kazabdu
+  - gaievski
+date: 2025-03-31
+has_science_table: true
+categories:
+  - technical-posts
+meta_keywords: z score normalization, OpenSearch 3.0-beta, neural search plugin, hybrid search, relevance ranking, search normalization, k-nn search, L2 normalization, how reciprocal rank fusion works
+meta_description: Learn about z score normalization using the Neural Search plugin in OpenSearch 3.0-beta. Discover how this new approach to hybrid search merges results from multiple query sources for improved relevance.
+---
+
+In the world of search engines and machine learning, data normalization plays a crucial role in ensuring fair and accurate comparisons between different features or scores. 
+Hybrid query uses multiple normalization techniques for preparing final results, main two types are score based normalization and rank base combination. In score base normalization, min-max normalization doesn’t work well with outliers (Outliers are data points that significantly differ from other observations in a dataset. 
+In the context of normalization techniques like Min-Max scaling and Z-score (Standard Score) normalization, outliers can have a substantial impact on the results). In this blogpost we would introduce another normalization technique called as z-score which was added in OpenSearch 3.0.0-beta release. 
+Let's dive into what Z-score normalization is, why it's important, and how it's being used in OpenSearch.
+
+## What is Z-Score Normalization?
+
+Z-score normalization, also known as standardization, is a method of scaling data using mean and standard deviation. The formula for calculating the Z-score is:
+Z = (X - μ) / σ
+Where:
+
+* X is the original value
+* μ is the mean of the population
+* σ is the standard deviation of the population
+
+## When to use Z Score?
+
+Considering your index’s structure can help you decide which one to choose since each has advantages of its own. If your documents are more similar to one another and the top-k results of a typical query return documents that are very similar to one another and clustered together within the index, as seen in the graph below, Min-Max may be a better option.
```

Comment:
Same as above, is there any way to compare to l2 instead of purely comparing against min-max?

Reply:
Replied above